python/ray/data/tests/test_dataset.py

```diff
@@ -5455,7 +5455,10 @@ def f(x):
         .fully_executed()
     )
 
-    assert f"{max_size}/{max_size} blocks" in ds.stats()
+    # The "max_size" parameter is taking effect in the legacy
+    # ActorPoolStrategy only.
+    if not DatasetContext.get_current().new_execution_backend:
+        assert f"{max_size}/{max_size} blocks" in ds.stats()
```

Comment:
Hmm isn't this a regression?

Reply:
I think this is missing feature in the new actor pool, which hasn't implemented the feature of capping the bundle size yet, @clarkzinzow 
Added TODO (as what we handled the missing feature of autoscaling) for now.

Reply:
Actually I think we should implement this capping feature since it's simpler and to control number of gaps.

Reply:
OK to leave this as a TODO (can add it to this issue: https://github.com/ray-project/ray/issues/31723)